datasources/beeper/beeper.go

```diff
@@ -0,0 +1,982 @@
+					rows.Close()
+					return fmt.Errorf("scanning row: %w", err)
+				}
+
+				// Extract data based on our knowledge of the Beeper schema
+				// Find important column indices
+				eventIDIdx := findColumnIndex(columns, "eventID")
+				roomIDIdx := findColumnIndex(columns, "roomID")
+				timestampIdx := findColumnIndex(columns, "timestamp")
+				messageIdx := findColumnIndex(columns, "message")
+				textContentIdx := findColumnIndex(columns, "text_content")
+				senderContactIDIdx := findColumnIndex(columns, "senderContactID")
+				isSentByMeIdx := findColumnIndex(columns, "isSentByMe")
+
+				// Extract the values
+				var eventID, roomID, messageText, senderContactID string
+				var timestamp time.Time
+				var isSentByMe bool
+
+				if eventIDIdx >= 0 && eventIDIdx < len(values) {
+					eventID = toString(values[eventIDIdx])
+				}
+
+				if roomIDIdx >= 0 && roomIDIdx < len(values) {
+					roomID = toString(values[roomIDIdx])
+				}
+
+				if timestampIdx >= 0 && timestampIdx < len(values) {
+					timestamp = toTimestamp(values[timestampIdx])
+				} else {
+					// Default to current time if no timestamp
+					timestamp = time.Now()
+				}
+
+				// Try to get text content from either message JSON or text_content field
+				if textContentIdx >= 0 && textContentIdx < len(values) {
+					messageText = toString(values[textContentIdx])
+				}
+
+				if messageText == "" && messageIdx >= 0 && messageIdx < len(values) {
+					messageJSON := toString(values[messageIdx])
+					// Try to extract text from JSON if possible
+					if strings.Contains(messageJSON, "\"body\":") {
+						// This is a very simple JSON extraction - in a real implementation
+						// you would properly parse the JSON
+						parts := strings.Split(messageJSON, "\"body\":")
+						if len(parts) > 1 {
+							bodyPart := parts[1]
+							if strings.Contains(bodyPart, "\"") {
+								bodyPart = strings.Split(bodyPart, "\"")[1]
+								messageText = bodyPart
+							}
+						}
+					}
+				}
+
+				if senderContactIDIdx >= 0 && senderContactIDIdx < len(values) {
+					senderContactID = toString(values[senderContactIDIdx])
+				}
+
+				if isSentByMeIdx >= 0 && isSentByMeIdx < len(values) {
+					isSentByMeVal := values[isSentByMeIdx]
+					switch v := isSentByMeVal.(type) {
+					case int64:
+						isSentByMe = v > 0
+					case bool:
+						isSentByMe = v
+					default:
+						isSentByMe = false
+					}
+				}
+
+				// Use threadID/roomID as chat name
+				chat := roomID
+				if chat == "" {
+					chat = "Beeper Chat"
+				}
+
+				// Use sender if available
+				sender := senderContactID
+				if sender == "" {
+					if isSentByMe {
+						sender = "Me"
+					} else {
+						sender = "Unknown"
+					}
+				}
+
+				// Create metadata map
+				metadata := timeline.Metadata{
+					"Source":     "Beeper",
+					"ChatApp":    "beeper",
+					"Chat":       chat,
+					"Sender":     sender,
+					"MessageID":  eventID,
+					"IsSentByMe": isSentByMe,
```

Comment:
Metadata keys can be human-friendly, i.e. "Chat app"  and "Is sent by me" and "Message ID".

That said, metadata like "Is sent by me" seems redundant, since the item's owner and relations should tell who sent it to whom. Similar with "Sender" and "MessageID" (since items have an ID field you can populate, I presume that is the same thing?).

Also, if the message is part of a specific chat, instead of "Chat" metadata, you can probably use a collection of sorts. See #81 for relevant discussion.

"Source" also seems pointless if it's always "Beeper".

So actually I think all these metadata fields might be unnecessary...